tortoise/tortoise.go

```diff
@@ -402,12 +403,19 @@ func (t *turtle) verifyLayers() {
 			}
 			success = t.full.verify(logger, target)
 		}
+
+		layer := t.layer(target)
 		if !success {
+			// notify mesh in two additional cases:
+			// - if layer was verified, and became undecided
+			// - if layer is undecided outside hdist distance
```

Comment:
do you think this will cause unnecessary churn? usually if the best cause block is executed in-place, and only reverted when it was verified invalid.

Reply:
in best case this condition has no impact, i doublechecked in [TestSmeshing](https://grafana.spacemesh.dev/explore?orgId=1&left=%7B%22datasource%22:%22loki-gke-dev%22,%22queries%22:%5B%7B%22refId%22:%22A%22,%22datasource%22:%7B%22type%22:%22loki%22,%22uid%22:%22loki-gke-dev%22%7D,%22editorMode%22:%22code%22,%22expr%22:%22%7Bnamespace%3D%5C%22test-cgdz%5C%22,%20pod%3D%5C%22smesher-11-55996f6664-8snm4%5C%22%7D%20%7C%3D%20%5C%22consensus%20results%5C%22%22,%22queryType%22:%22range%22%7D%5D,%22range%22:%7B%22from%22:%221685699059384%22,%22to%22:%221685700055021%22%7D%7D)

it will have to do more work than optimal if there is a partition that causes verified layer to go backward (e.g we knew about 100 weight, and then got 100 more weight). but it is still ok, as we already applied necessary block before doing expensive operations